src/module/data/item/ability.mjs

```diff
@@ -30,9 +32,10 @@ export default class AbilityModel extends BaseItemModel {
     schema.type = new fields.StringField(requiredChoice(config.types, "action"));
     schema.distance = new fields.SchemaField({
       type: new fields.StringField(requiredChoice(config.distances, "self")),
-      primary: new fields.NumberField(),
-      secondary: new fields.NumberField()
+      primary: new fields.NumberField({integer: true, required: true}),
+      secondary: new fields.NumberField({integer: true, required: true})
```

Comment:
I don't believe `required: true` does anything for a NumberField; it cannot be `undefined`.